src/tcp.rs

```diff
@@ -0,0 +1,50 @@
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+
+/// Configuration for keepalive probes in a TCP stream.
+///
+/// This config's properties map to TCP keepalive properties in Tokio/Mio:
+/// https://github.com/tokio-rs/mio/blob/e6e403fe2a4fc14dfbc74dbb3ae3a14e3044eb6f/src/net/tcp/socket.rs#L25-L46
+///
+/// # Note
+///
+/// Support for the `interval` and `retries` options has just landed in Mio and they are not
```

Comment:
I might just drop these unimplemented options from the below struct to reduce confusion.

Reply:
Done in 59c042f.